client/src/main/proto/spine/validation.proto

```diff
@@ -57,43 +56,173 @@ extend google.protobuf.FieldOptions {
+// TODO:2016-03-25:alexander.litus: error messages format
 
+// If the field type is `message`, it must be set to a non-default instance;
+// if it is `string` or `bytes`, the value must not be an empty string or array.
+// Other field types are not applicable.
+// If the field is repeated, it must have at least one value set, and all its values must be valid.
+//
+// Example:
+//
+// MyMessage field = 1 [(required).is = true];
+//
 message RequiredOption {
-    // The default format string for missing field. The parameter is the name of the missing field.
-    option (default_message) = "Field %s must be set";
+    // The default error message format string.
+    // The format parameter is the simple name of the missing field.
+    option (default_message) = "'%s' must be set";
 
-    bool value = 1;
+    // Specifies if the field is required. Must be set to true. Otherwise, the field is not validated.
+    // The default value is false.
+    bool is = 1;
 
-    // User-defined validation error message format.
+    // A user-defined validation error format message.
+    // The format parameters are the same as for the default message.
     string msg = 2;
 }
 
+// TODO:2016-03-14:alexander.litus: is this option really needed? Why not use min/max instead?
+// Applicable only to numbers.
 message DecimalOption {
-    option (default_message) = "Field value must fit into range";
+    option (default_message) = "field value must fit into range";
 
-    int32 min = 1;
-    int32 max = 2;
+    double min = 1;
+    double max = 2;
     bool inclusive = 3;
     string msg = 4;
 }
 
-message DigitsOption {
-    option (default_message) = "Field should have valid digital value";
+// The field value must be greater than (or equal to, if the `inclusive` parameter is true) the given minimum number.
+// Is applicable only to numbers.
+// Repeated fields are supported.
+// This option is considered as not set only if both min and max options are not set,
+// or if the `ignore` parameter is set to true.
+//
+// Example:
+//
+// double value = 1 [(min).is = 10.5, (min).inclusive = true];
+//
+// TODO:2016-03-25:alexander.litus: change options
+message MinOption {
+    // The default error message format string.
+    // The format parameters are:
+    // 1) the simple name of the field;
+    // 2) "or equal to " string (if the `inclusive` parameter is true);
+    // 3) the minimum number;
+    // 4) an actual value of the field.
+    option (default_message) = "'%s' must be greater than %s%s, actual: %s";
+
+    // The minimum value of the field.
+    // The default value is zero.
+    double is = 1;
+
+    // Specifies if the field can be equal to the minimum value.
+    // The default value is false.
+    bool inclusive = 2;
+
+    // Set this property to true if it is needed to use only the `max` option.
+    // The default value is false.
+    bool ignore = 3;
```

Comment:
Assume if `max` is not specified, then there's no maximum, if the minimum value < 0.  If minimum is less than zero, than zero is the maximum, even if it's not specified.